privacy/scrubber.go

```diff
@@ -81,6 +82,63 @@ func NewScrubber() Scrubber {
 	return scrubber{}
 }
 
+func (scrubber) ScrubRequest(bidRequest *openrtb2.BidRequest, enforcement Enforcement) *openrtb2.BidRequest {
+	if enforcement.UFPD {
+		// transmitUfpd covers user.ext.data, user.data, user.id, user.buyeruid, user.yob, user.gender, user.keywords, user.kwarray
+		// and device.{ifa, macsha1, macmd5, dpidsha1, dpidmd5, didsha1, didmd5}
+		if bidRequest.Device != nil {
+			bidRequest.Device.DIDMD5 = ""
+			bidRequest.Device.DIDSHA1 = ""
+			bidRequest.Device.DPIDMD5 = ""
+			bidRequest.Device.DPIDSHA1 = ""
+			bidRequest.Device.IFA = ""
+			bidRequest.Device.MACMD5 = ""
+			bidRequest.Device.MACSHA1 = ""
+		}
+		if bidRequest.User != nil {
+			bidRequest.User.Data = nil
+			bidRequest.User.Ext = scrubExtIDs(bidRequest.User.Ext, "data")
+			bidRequest.User.ID = ""
+			bidRequest.User.BuyerUID = ""
+			bidRequest.User.Yob = 0
+			bidRequest.User.Gender = ""
+			bidRequest.User.Keywords = ""
+			bidRequest.User.KwArray = nil
+		}
+	}
+	if enforcement.Eids {
+		//transmitEids covers user.eids and user.ext.eids
+		if bidRequest.User != nil {
+			bidRequest.User.EIDs = nil
+			bidRequest.User.Ext = scrubExtIDs(bidRequest.User.Ext, "eids")
+		}
+	}
+
+	if enforcement.TID {
+		//remove source.tid and imp.ext.tid
+		if bidRequest.Source != nil {
+			bidRequest.Source.TID = ""
+		}
+		for ind, imp := range bidRequest.Imp {
+			impExt := scrubExtIDs(imp.Ext, "tid")
```

Comment:
It's possible for `scrubExtIDs` to be called 3 times in this method, each time performing a marshal and unmarshal. Please optimize so there is at most just one marshal and unmarshal to keep performance in check.

Reply:
Yes, with the current approach UserExt may be scrubbed twice. In order to prevent this we need to pass `enforcement.Eids` to `ScrubUser`. 
In the future we will replace actual request with RequestWrapper and I believe it will solve this problem. 
Do you think we can go with current approach temporarily?